src/background.js

```diff
@@ -29,50 +29,74 @@ function sendVideoUrlToMetubeAndSwitchTab(videoUrl, metubeUrl, tab) {
         if (res.ok === true && res.status === 200) {
             return res.json();
         }
-        // todo fix it
-        alert("error :: code " + res.status);
+        console.log("error :: code" + res.status);
     }).then(function(result) {
-        if (result.status === "ok") {
-            openTab(metubeUrl, tab);
-        } else {
-            // todo fix it
-            alert("error :: " + json);
+        if (result.status !== "ok") {
+            console.log("error :: ", result );
+            if (result.msg.includes('URLError')) {
+                // Go straight to catch block and skip the callback
+                throw new Error('Error when adding URL to MyTube');
+            }
         }
-    }).catch(function(res) {
-        alert("error :: " + res);
+    }).then(function() {
+        typeof callback === 'function' && callback();
+    }).catch(function(e) {
+        console.log("Ran into an error :: ", e);
     });
 }
 
 chrome.contextMenus.onClicked.addListener(function(item, tab) {
-    chrome.storage.sync.get(['metube'], function(data) {
+    chrome.storage.sync.get(['metube', 'contextMenuClickBehavior'], function(data) {
         if (data === undefined || !data.hasOwnProperty('metube') || data.metube === "") {
             openTab(chrome.runtime.getURL('options.html'), tab);
             return
         }
-        sendVideoUrlToMetubeAndSwitchTab(item.linkUrl, data.metube, tab);
+        if ( data.contextMenuClickBehavior == 'context-menu-send-current-url') {
+            sendVideoUrlToMetube(item.linkUrl, data.metube);
+        } else if (data.contextMenuClickBehavior == 'context-menu-send-current-url-and-switch'){
+            sendVideoUrlToMetube(item.linkUrl, data.metube, function() {
+                openTab(data.metube, tab);
+            });
+        } else {
+            console.log("Unknown contextMenuClickBehavior value: " + data.contextMenuClickBehavior);
```

Comment:
in what condition that `else` is possible? maybe remove that and leave 
```
sendVideoUrlToMetube(item.linkUrl, data.metube, function() {
   openTab(data.metube, tab);
});
```
as default for `else` block?

Reply:
I've coded the default contextMenuClickBehavior value as a checked radio box inside the options.html file [here](https://github.com/Rpsl/metube-browser-extension/pull/7/files#diff-70b26bb50b16514f30f51e300930a4e23d2acfb8bbea7ece91a3ed037c98c07aR45) 

I could follow your suggestion, but that would mean changing the default behavior at a later time would mean a change in the options.html page (to change which radio button is pre-checked for the user) and also this code here.

The way this code is, it allows you to change the default behavior by only changing which radio button is checked by default and that's all.
If a programmer somehow removes the default pre-checked option from options.html, then this final else will trigger in that case and that case alone. A user will never be able to trigger this else block.

Reply:
I don't agree with this. You are trying to make the code more complicated from situations that don't exist. 

please, let's rewrite this block of code to keep it as simple as possible


```javascript
let needToSwitch = (data.contextMenuClickBehavior === 'context-menu-send-current-url-and-switch');

sendVideoUrlToMetube(item.linkUrl, data.metube, function() {
    if(!needToSwitch){
        return;
    }
    openTab(data.metube, tab);
});
```

Reply:
Done - added needToSwitch variable and simplified some code.